File: app/src/main/java/sensors_in_paradise/xsens/MainActivity.java
package sensors_in_paradise.xsens;

import androidx.appcompat.app.AppCompatActivity;

import android.bluetooth.BluetoothDevice;
import android.bluetooth.le.ScanSettings;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.xsens.dot.android.sdk.XsensDotSdk;
import com.xsens.dot.android.sdk.events.XsensDotData;
import com.xsens.dot.android.sdk.interfaces.XsensDotDeviceCallback;
import com.xsens.dot.android.sdk.interfaces.XsensDotScannerCallback;
import com.xsens.dot.android.sdk.models.FilterProfileInfo;
import com.xsens.dot.android.sdk.utils.XsensDotScanner;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements XsensDotDeviceCallback, XsensDotScannerCallback {
    TextView tv;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tv = findViewById(R.id.textView);

        XsensDotSdk.setDebugEnabled(true);
        initXsScanner();
        boolean scanStarted = mXsScanner.startScan();

        Log.println(Log.INFO,"XSENS", "");
        tv.setText("Scan started: "+scanStarted);

    }
    private XsensDotScanner mXsScanner;
    private void initXsScanner() {
        mXsScanner = new XsensDotScanner(this, this);
        mXsScanner.setScanMode(ScanSettings.SCAN_MODE_BALANCED);
    }

    @Override
    public void onXsensDotConnectionChanged(String s, int i) {

    }

    @Override
    public void onXsensDotServicesDiscovered(String s, int i) {

    }

    @Override
    public void onXsensDotFirmwareVersionRead(String s, String s1) {

    }

    @Override
    public void onXsensDotTagChanged(String s, String s1) {

    }

    @Override
    public void onXsensDotBatteryChanged(String s, int i, int i1) {

    }

    @Override
    public void onXsensDotDataChanged(String s, XsensDotData xsensDotData) {

    }

    @Override
    public void onXsensDotInitDone(String s) {

    }

    @Override
    public void onXsensDotButtonClicked(String s, long l) {

    }

    @Override
    public void onXsensDotPowerSavingTriggered(String s) {

    }

    @Override
    public void onReadRemoteRssi(String s, int i) {

    }

    @Override
    public void onXsensDotOutputRateUpdate(String s, int i) {

    }

    @Override
    public void onXsensDotFilterProfileUpdate(String s, int i) {

    }

    @Override
    public void onXsensDotGetFilterProfileInfo(String s, ArrayList<FilterProfileInfo> arrayList) {

    }

    @Override
    public void onSyncStatusUpdate(String s, boolean b) {

    }

    @Override
    public void onXsensDotScanned(BluetoothDevice device, int i) {
        String name = device.getName();
        String address = device.getAddress();

        tv.setText("Name: "+name+" address:"+address);
        Log.println(Log.INFO,"XSENS", "Device discovered: "+name+" ---> "+address);
    }
}

File: app/src/main/java/sensors_in_paradise/xsens/MainActivity.kt
package sensors_in_paradise.xsens

import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.*
import com.google.android.material.tabs.TabLayout
import com.xsens.dot.android.sdk.events.XsensDotData
import com.xsens.dot.android.sdk.models.XsensDotDevice
import sensors_in_paradise.xsens.page1.ConnectionInterface
import sensors_in_paradise.xsens.page1.Page1Handler
import sensors_in_paradise.xsens.page2.Page2Handler
import sensors_in_paradise.xsens.page1.XSENSArrayList

class MainActivity : AppCompatActivity(), TabLayout.OnTabSelectedListener, ConnectionInterface {

    private lateinit var flipper: ViewFlipper
    private lateinit var tabLayout: TabLayout
    private lateinit var devices: ArrayList<XsensDotDevice>

    private val pageHandlers = ArrayList<PageInterface>()

    private val scannedDevices = XSENSArrayList()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        flipper = findViewById(R.id.flipper_activity_main)
        tabLayout = findViewById(R.id.tab_layout_activity_main)

        initClickListeners()

        var page2 = Page2Handler(scannedDevices)
        pageHandlers.add(page2)
        pageHandlers.add(Page1Handler(scannedDevices,page2))

        for (handler in pageHandlers) {
            handler.activityCreated(this)
        }
        supportActionBar?.setBackgroundDrawable(ColorDrawable(getColor(R.color.colorPrimary)))
    }

    override fun onResume() {
        super.onResume()
        for (handler in pageHandlers) {
            handler.activityResumed()
        }
    }
    private fun initClickListeners() {
        tabLayout.addOnTabSelectedListener(this)
    }
    override fun onTabSelected(tab: TabLayout.Tab?) {
        if (tab != null) {
            flipper.displayedChild = tab.position
        }
    }

    override fun onTabUnselected(tab: TabLayout.Tab?) {
        //TODO("Not yet implemented")
    }

    override fun onTabReselected(tab: TabLayout.Tab?) {
        //TODO("Not yet implemented")
    }

    override fun onConnectedDevicesChanged(deviceAddress: String, connected: Boolean) {
        //TODO("Not yet implemented")
    }

    override fun onXsensDotDataChanged(deviceAddress: String, xsensDotData: XsensDotData) {
        //TODO("Not yet implemented")
    }

    override fun onXsensDotOutputRateUpdate(deviceAddress: String, outputRate: Int) {
        //TODO("Not yet implemented")
    }
}
